File: src/main/java/ordilov/randomplay/like/domain/LikeStore.java
package ordilov.randomplay.like.domain;

public interface LikeStore {
  LikedTrack store(LikedTrack likedTrack);
  LikedPlaylist store(LikedPlaylist likedPlaylist);
  void delete(LikedTrack likedTrack);
  void delete(LikedPlaylist likedPlaylist);
}

File: src/main/java/ordilov/lissn/like/domain/LikeStore.kt
package ordilov.lissn.like.domain

interface LikeStore {

    fun store(likedTrack: LikedTrack): LikedTrack

    fun store(likedPlaylist: LikedPlaylist): LikedPlaylist

    fun delete(likedTrack: LikedTrack)

    fun delete(likedPlaylist: LikedPlaylist)
}